File: src/main/java/ai/aimachineserver/application/commands/CreateUserCommand.java
package ai.aimachineserver.application.commands;

import lombok.AllArgsConstructor;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

@AllArgsConstructor
public class CreateUserCommand {
    @NotNull(message = "username cannot be null")
    @Size(min = 1, max = 100)
    public String username;

    @NotNull(message = "password cannot be null")
    @Size(min = 1, max = 100)
    public String password;
}

File: src/main/kotlin/ai/aimachineserver/application/commands/CreateUserCommand.kt
package ai.aimachineserver.application.commands

import javax.validation.constraints.NotNull
import javax.validation.constraints.Size

class CreateUserCommand(
    @NotNull(message = "username cannot be null")
    @Size(min = 1, max = 100)
    val username: String,

    @NotNull(message = "password cannot be null")
    @Size(min = 1, max = 100)
    val password: String
) {
    private constructor() : this("", "")
}
